docsk8s/setting-up/ls-k8s-secure.asciidoc

```diff
@@ -5,19 +5,234 @@ WARNING: This documentation is still in development and may be changed or remove
+apiVersion: beat.k8s.elastic.co/v1beta1
+kind: Beat
+metadata:
+  name: demo
+spec:
+  type: filebeat
+  config:
+    output.logstash:
+      ssl.certificate_authorities: ["/usr/share/filebeat/ca.crt"]
+      ssl.certificate: "/usr/share/filebeat/client.crt"
+      ssl.key: "/usr/share/filebeat/client.key"
+(...)
+  deployment:
+    podTemplate:
+      spec:
+        containers:
+          - name: filebeat
+            volumeMounts:
+              - name: logstash-beats-tls
+                mountPath: /usr/share/filebeat/ca.crt
+                subPath: ca.crt
+              - name: logstash-beats-tls
+                mountPath: /usr/share/filebeat/client.key
+                subPath: client.key
+              - name: logstash-beats-tls
+                mountPath: /usr/share/filebeat/client.crt
+                subPath: client.crt
+        volumes:
+          - name: logstash-beats-tls
+            secret:
+              secretName: logstash-beats-tls
+--
+
+[[security-k8s-secret]]
+==== Using secret
+
+One way to config confidential data, such as password, is to use `Secret` and references it as container environment variable.
 
+Encode confidential data with Base64:
+
+[source,sh]
+--
+echo -n "changeme" | base64
+--
+
+Create `Secret` to hold the result of the encoding:
+
+[source,yaml]
+--
+apiVersion: v1
+kind: Secret
+metadata:
+  name: logstash-secret
+type: Opaque
+data:
+  ES_PW: Y2hhbmdlbWU=
+--
+
+Reference the confidential data in `Deployment`:
+
+[source,yaml]
+--
+    spec:
+      containers:
+        - name: logstash
+          env:
+            - name: ELASTICSEARCH_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: logstash-secret
+                  key: ES_PW
+--
+
+[[security-logstash-keystore]]
 ==== Using the {ls} keystore
-Description.
 
+{ls} can use the key of {logstash-ref}/keystore.html[keystore] in place of the confidential data when configure sensitive settings.
+
+To create `Secret` from an existing keystore `logstash.keystore`:
+
+[source,sh]
+--
+kubectl create secret generic logstash-keystore --from-file=logstash.keystore --dry-run=client -o yaml
+--
+
+Mount the `Secret` to the {ls} config directory in `Deployment`:
+
+[source,yaml]
+--
+apiVersion: apps/v1
+kind: Deployment
+(...)
+    spec:
+      containers:
+        - name: logstash
+          env:
+            - name: LOGSTASH_KEYSTORE_PASS <1>
+              valueFrom:
+                secretKeyRef:
+                  name: logstash-secret
+                  key: LOGSTASH_KEYSTORE_PASS
+(...)
+          volumeMounts:
+            - name: logstash-keystore
+              mountPath: /usr/share/logstash/config/logstash.keystore
+              subPath: logstash.keystore
+      volumes:
+        - name: logstash-keystore
+          secret:
+            secretName: logstash-keystore
+--
+
+<1> `LOGSTASH_KEYSTORE_PASS` is required only when the keystore is protected by {logstash-ref}/keystore.html#keystore-password[password]
```

Comment:
Maybe remove "only", as we recommend the use of password protection in production

```suggestion
<1> `LOGSTASH_KEYSTORE_PASS` is required when the keystore is protected by {logstash-ref}/keystore.html#keystore-password[password]
```